src/main.cc

```diff
@@ -92,38 +139,20 @@ int main(int argc, char *argv[]) {
     const char *cmd_list = "Available commands: SendDeviceData, CheckUpdates, Download, Install, CampaignCheck, CampaignAccept, Pause, Resume, Abort";
     std::cout << cmd_list << std::endl;
 
-    std::vector<Uptane::Target> current_updates;
-    std::string buffer;
-    while (std::getline(std::cin, buffer)) {
-      std::vector<std::string> words;
-      boost::algorithm::split(words, buffer, boost::is_any_of("\t "), boost::token_compress_on);
-      std::string &command = words[0];
-      std::string &param = words[1];
-      boost::algorithm::to_lower(command);
-      if (command == "senddevicedata") {
-        aktualizr.SendDeviceData().get();
-      } else if (command == "checkupdates") {
-        auto result = aktualizr.CheckUpdates().get();
-        current_updates = result.updates;
-      } else if (command == "download") {
-        aktualizr.Download(current_updates).get();
-      } else if (command == "install") {
-        aktualizr.Install(current_updates).get();
-        current_updates.clear();
-      } else if (command == "campaigncheck") {
-        aktualizr.CampaignCheck().get();
-      } else if (command == "campaignaccept") {
-        aktualizr.CampaignControl(param, campaign::Cmd::Accept).get();
-      } else if (command == "pause") {
-        aktualizr.Pause();
-      } else if (command == "resume") {
-        aktualizr.Resume();
-      } else if (command == "abort") {
-        aktualizr.Abort();
-      } else if (!command.empty()) {
+    if (commandline_map.count("execute")) {
+      LOG_INFO << "REPL mode disabled";
+      if (!process_command(&aktualizr, commandline_map["execute"].as< std::string >())) {
```

Comment:
Will this work with things that require an argument?

Reply:
No, it won't. I think so far it's only one command related to campaings which requires an argument - I've not used it so I've opted to avoid extra complexity. Cool idea for future improvement though.

Reply:
I think if this is meant to be a nice demo app to help new users get comfortable with the API, we shouldn't recommend shortcuts that don't work consistently. I think we should address this issue with the extra parameter for campaigns before merging.